Sources/Bot/Services/MergeService.swift

```diff
@@ -401,6 +405,45 @@ extension MergeService {
 
 // MARK: - System types
 
+extension MergeService {
+    public final class Healthcheck {
+
+        public enum Reason: Error, Equatable {
+            case potentialDeadlock
+        }
+
+        public enum Status: Equatable {
+            case ok
+            case unhealthy(Reason)
+        }
+
+        public let status: Property<Status>
+
+        internal init(
+            state: Signal<State, NoError>,
+            statusChecksTimeout: TimeInterval,
+            scheduler: DateScheduler
+            ) {
+            status = Property(
+                initial: .ok,
+                then: state.combinePrevious()
+                    .skipRepeats { lhs, rhs in
+                        return lhs == rhs
+                    }
```

Comment:
I'm not a RAS expert, but wouldn't `state.skipRepeats()` be enough there instead of `state.combinePrevious().skipRepeats { … == … }`? I mean I thought `skipRepeats` would do that for you already

Reply:
We are working with tuples here and in Swift 4.2 they not `Equatable` out of the box.

Reply:
Oh 😢